synapse/federation/federation_client.py

```diff
@@ -405,6 +405,9 @@ async def get_room_state_ids(
 
         Returns:
             a tuple of (state event_ids, auth event_ids)
+
+        Raises:
+            InvalidResponseError: if fields in the response have the wrong type.
```

Comment:
this probably needs chasing up the call stack (through `_get_state_ids_after_missing_prev_event`, `_resolve_state_at_missing_prevs`, etc) too.